tests/integration/security/egress_sidecar_tls_origination_test.go

```diff
@@ -0,0 +1,203 @@
+//go:build integ
+// +build integ
+
+// Copyright Istio Authors
+//
+// Licensed under the Apache License, Version 2.0 (the "License");
+// you may not use this file except in compliance with the License.
+// You may obtain a copy of the License at
+//
+//     http://www.apache.org/licenses/LICENSE-2.0
+//
+// Unless required by applicable law or agreed to in writing, software
+// distributed under the License is distributed on an "AS IS" BASIS,
+// WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+// See the License for the specific language governing permissions and
+// limitations under the License.
+
+package security
+
+import (
+	"net/http"
+	"path"
+	"strings"
+	"testing"
+
+	"istio.io/istio/pkg/config/protocol"
+	"istio.io/istio/pkg/http/headers"
+	"istio.io/istio/pkg/test/env"
+	"istio.io/istio/pkg/test/framework"
+	"istio.io/istio/pkg/test/framework/components/echo"
+	"istio.io/istio/pkg/test/framework/components/echo/check"
+	"istio.io/istio/pkg/test/framework/components/namespace"
+	"istio.io/istio/pkg/test/framework/resource/config/apply"
+	"istio.io/istio/pkg/test/util/file"
+	ingressutil "istio.io/istio/tests/integration/security/sds_ingress/util"
+	sdstlsutil "istio.io/istio/tests/integration/security/sds_tls_origination/util"
+)
+
+// TestMutualTlsOrigination test MUTUAL TLS mode with TLS origination happening at Sidecar
+// It uses CredentialName set in DestinationRule API to fetch secrets from k8s API server
+func TestSidecarMutualTlsOrigination(t *testing.T) {
+	// nolint: staticcheck
+	framework.NewTest(t).
+		Features("security.egress.mtls.sds").
+		Run(func(t framework.TestContext) {
+			var (
+				credNameGeneric = "mtls-credential-generic"
+				fakeCredName    = "fake-mtls-credential"
+			)
+
+			// Create a valid kubernetes secret to provision key/cert for sidecar.
+			ingressutil.CreateIngressKubeSecretInNamespace(t, credNameGeneric, ingressutil.Mtls, ingressutil.IngressCredential{
+				Certificate: file.AsStringOrFail(t, path.Join(env.IstioSrc, "tests/testdata/certs/dns/cert-chain.pem")),
+				PrivateKey:  file.AsStringOrFail(t, path.Join(env.IstioSrc, "tests/testdata/certs/dns/key.pem")),
+				CaCert:      file.AsStringOrFail(t, path.Join(env.IstioSrc, "tests/testdata/certs/dns/root-cert.pem")),
+			}, true, apps.Ns1.Namespace.Name())
+
+			// Authorize only one of the sidecars in the namespace with secret listing permissions.
+			serviceAccount := apps.Ns1.A.Config().ServiceAccountName()
+			serviceAccountName := serviceAccount[strings.LastIndex(serviceAccount, "/")+1:]
```

Comment:
isn't a method that can get sa directly

Reply:
https://github.com/istio/istio/blob/master/pkg/test/framework/components/echo/config.go#L204 This is the only one currently available. Did not add a utility thinking only this test needs this.

Reply:
IC